elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -79,6 +119,117 @@ private void logContext(RelationshipOp operation, Environment environment) {
     }
 
+    /**
+     * handle FETCH operation
+     * @param context environment encapsulating graphQL's request environment
+     * @return list of persistent resource objects
+     */
+    private Object fetchObjects(Environment context) {
+        /* sanity check for data argument w FETCH */
+        if(context.data.isPresent()) {
+            throw new BadRequestException("FETCH must not include data");
+        }
+
+        /* check whether current object has a parent or not */
+        if (context.isRoot() || context.outputType instanceof GraphQLList) {
+            /* top level entity */
+            GraphQLObjectType graphQLType = (GraphQLObjectType) ((GraphQLList) context.outputType).getWrappedType();
+            String entityType = graphQLType.getName();
+            Class recordType = context.requestScope.getDictionary().getEntityClass(entityType);
+
+            if (recordType == null) {
+                throw new UnknownEntityException(entityType);
+            }
+
+            return fetchObject(context.requestScope, recordType, context.ids, context.sort,
+                    context.offset, context.first, context.filters);
+        } else {
+            /* fetch properties */
+            return fetchObject(context.requestScope, context.parentResource, context.field,
+                    context.outputType, context.ids, context.sort, context.offset, context.first, context.filters);
+        }
+    }
+
+    /**
+     * FETCH top level entity either by ids or as a collection
+     * @param requestScope request scope
+     * @param recordType record type
+     * @param ids list of ids (can be NULL)
+     * @param sort sort by ASC/DESC
+     * @param offset pagination offset argument
+     * @param first pagination first argument
+     * @param filters filter params
+     * @return list of persistent resource objects
+     */
+    private Object fetchObject(RequestScope requestScope, Class recordType, Optional<List<String>> ids,
+                               Optional<String> sort, Optional<String> offset,
+                               Optional<String> first, Optional<String> filters) {
+        /* fetching a collection */
+        if(!ids.isPresent()) {
+            List<PersistentResource> records = new ArrayList<>(PersistentResource.
+                    loadRecords(recordType, requestScope));
+
+            //TODO: paginate/filter/sort
+            return records;
+
+        } else { /* fetching by id */
+            List<String> idList = ids.get();
+
+            /* handle empty list of ids */
+            if(idList.isEmpty()) {
+                throw new IllegalArgumentException("Empty list passed to ids");
+            }
+
+            /* access records from internal db and return */
+            HashSet recordSet = new HashSet();
+            for (String id : idList) {
+                recordSet.add(PersistentResource.loadRecord(recordType, id, requestScope));
```

Comment:
This will result in a bunch of round trip calls to the datastore/DB.  A better approach would be to add the list of IDS as filters and then just call loadRecords with the list of filters.